### DISCUSSION TITLE
Operation with Gen2 Goodwe inverters

### DISCUSSION THREAD

**Discussion by DezR** [ArminJo/JK-BMSToPylontechCAN#32] [General]
Hey Arminjo,
I've loaded a Nano with your code and connected between a JK BD4A17S4P and a Goodwe GW5000-ES-20 inverter, however the inverter always has the battery staying in Standby (or Spare) mode with no charge or discharge.
There are several packs with BD4A17S4P's paralleled, but just trying to get 1 BMS working first.

Attached is a dump of how the inverter sees the battery using inverter_check.py from https://github.com/marcelblijleven/goodwe  

[inverter_check.txt](https://github.com/ArminJo/JK-BMSToPylontechCAN/files/14398794/inverter_check.txt)

The inverter is currently configured to be using a Pylon US3000 (x1) battery, though get similar results with any Pylon profile on this inverter. 
The battery status doesn't change even when using 'self-use', 'general' or 'off-grid' modes on the inverter.

Is there a potential (or known) problem using your code with newer generations of Goodwe inverters like ES Gen2 and ET series...or that type of JK BMS?

**Comment by ArminJo** [comment]
First you can look at the serial output of the program and press the button, which enables CAN debug output.
But maybe your inverter expexts another frame type not sent or generated already to work correct.
Or it is disturbed by the extension frames which is unlikely. The Deye inverter accepts / skips these frames.

**Comment by DezR** [comment]
Thank you.
The program serial output looks to be ok. I'll check the CAN debug output again.
There does seem to be something a bit different with the newest Goodwe inverters.

**Reply by ArminJo** [reply]
Any news?
